### PR TITLE
feat: INFRA-1724 split email transport to adapters

### PR DESCRIPTION
**Pull request by sitozzz**
## Summary by CodeRabbit

* **New Features**
  * Added transactional email delivery via **Mailgun** and **Unisender Go**, selectable through `EMAIL_API_CONFIG.type`.
  * Enhanced recipient parsing (To/CC/BCC), support for optional reply-to/tags/extended params, and attachment handling from URLs or buffers (with timeout and max-size limits).
* **Documentation**
  * Expanded notification domain docs with provider setup, required config examples, and guidance for adding new providers (including SPF/DKIM notes).
* **Refactor**
  * Updated email sending flows to use the new adapter layer consistently (including the resident phone email task).
* **Tests**
  * Added Jest coverage for configuration/availability checks, validation, provider request/response handling, and attachment edge cases.

### CODE REVIEW

**Review by dkoviazin — APPROVED** [review]

**Comment by sitozzz** [comment]
@coderabbitai review

File: apps/condo/domains/notification/adapters/emailAdapter.js [hunk 1]
```diff
@@ -0,0 +1,577 @@
...
+
+const conf = require('@open-condo/config')
+const { fetch } = require('@open-condo/keystone/fetch')
+const { getLogger } = require('@open-condo/keystone/logging')
+
+
+const logger = getLogger()
+
+const HTTPX_REGEXP = /^http:/
+const NAMED_EMAIL_REGEXP = /^(.+?)\s*<\s*([^>]+)\s*>$/
```
Comment (toplenboren) [inline apps/condo/domains/notification/adapters/emailAdapter.js]: check backtracking pls

**Review by toplenboren — APPROVED** [review]
lgtm
